model/registry/geo.yaml

```diff
@@ -0,0 +1,113 @@
+              value: 'AS'
+              brief: 'Asia'
+            - id: 'eu'
+              stability: experimental
+              value: 'EU'
+              brief: 'Europe'
+            - id: 'na'
+              stability: experimental
+              value: 'NA'
+              brief: 'North America'
+            - id: 'oc'
+              stability: experimental
+              value: 'OC'
+              brief: 'Oceania'
+            - id: 'sa'
+              stability: experimental
+              value: 'SA'
+              brief: 'South America'
+      - id: continent_name
+        stability: experimental
+        type: string
+        brief: >
+          Name of the continent.
+        examples: [ 'North America', 'Europe' ]
+      - id: country_iso_code
+        stability: experimental
+        type: string
+        brief: >
+          Two-letter ISO Country Code ([ISO 3166-1 alpha2](https://en.wikipedia.org/wiki/ISO_3166-1#Codes)).
+        examples: [ 'CA' ]
+      - id: country_name
+        stability: experimental
+        type: string
+        brief: >
+          Country name.
+        examples: [ 'Canada' ]
+      - id: location.lon
+        stability: experimental
+        type: double
+        brief: >
+          Longitude of the geo location in [WGS84](https://en.wikipedia.org/wiki/World_Geodetic_System#WGS84).
+        examples: [ -73.614830 ]
+      - id: location.lat
+        stability: experimental
+        type: double
+        brief: >
+          Latitude of the geo location in [WGS84](https://en.wikipedia.org/wiki/World_Geodetic_System#WGS84).
+        examples: [ 45.505918 ]
+      - id: name
+        stability: experimental
+        type: string
+        brief: >
+          User-defined description of a location.
+        note: >
+          User-defined description of a location, at the level of granularity they care about.
+          Could be the name of their data centers, the floor number, if this describes a local physical entity, city names.
+          Not typically used in automated geolocation.
+        examples: [ 'boston-dc' ]
+      - id: postal_code
+        stability: experimental
+        type: string
+        brief: >
+          Postal code associated with the location.
+          Values appropriate for this field may also be known as a postcode or ZIP code and will vary widely from country to country.
+        examples: [ '94040' ]
+      - id: region_iso_code
+        stability: experimental
+        type: string
+        brief: >
+          Region ISO code ([ISO 3166-2](https://en.wikipedia.org/wiki/ISO_3166-2)).
+        examples: [ 'CA-QC' ]
+      - id: region_name
+        stability: experimental
+        type: string
+        brief: >
+          Region name.
+        examples: [ 'Quebec' ]
+      - id: timezone
+        stability: experimental
+        type: string
+        brief: >
+          The time zone of the location, such as [IANA time zone name](https://nodatime.org/TimeZones).
```

Comment:
I think we should be more strict on which format we expect the timezone identifier. Also is just `timezone` enough? This I supposed is the identifier, but there's also for ex the tz abbreviation (GMT, CET). Don't we also want to have that? If so then this should also become a namespace.

Reply:
I think in this context forcing very accurate timezone reporting is super hard. So to me the question is: How much accuracy do we want to push here on implementors of this vs how much free room do we want to leave?

To me the point of this attribute would be to report the timezone of the event which was enriched with geo fields, so users know about the location of the event - which potentially includes the timezone of the location as well.

The potential problem I see with some of common timezone abbreviations is the daylight saving issue. If we have CET, then we'd need to also have CEST and if we limit this field, then implementations should deal with that. And this may force implementations to look at timestamp and figure out the current time zone name based on that.

But that's still not 100% clear - if an event was recorded in January and has `geo.timezone: CET`, but you look at the event in June, is it then correct? Or shouldn't it be `CEST`? Probably `CET` is correct, but are we 100% sure?

What I like about the [nodatime link](https://nodatime.org/TimeZones) is that it contains a list of official names and to me it seems you can (almost?) always populate `timezone` with a city based name (e.g. `Europe/Vienna`). This enables reporting the time-zone of the event in a format where the implementation does not need to care about daylight saving. In the context of geo, I feel this is the best way to go.

At the same time, it's still a string field, so if some implementation wants to go further and always report the common tz abbreviation, it can do so. But I agree, this introduces some ambiguity.

Given the context here, I'd vote for leaving it as is.

Alternatively we could also drop this field for now, unless someone has a specific use-case for it. I'd argue this field is actually redundant: I'd bet most (all?) implementations would just calculate the time-zone based on some other fields (e.g. `geo.location.lat/lon` or `geo.city.name`). Those are the more important fields here and this calculation can then happen later in the backend or even in the UI if someone wants to have a feature that shows the time-zone of the event.

Happy to hear other opinions.

Reply:
Removed this field for now and opened https://github.com/open-telemetry/semantic-conventions/issues/1213 to discuss.